lrauv_system_tests/test/vehicle/test_sensor_timeinterpolation.cc

```diff
@@ -0,0 +1,171 @@
+  spawnMsg.set_initlat_(lat1.Degree());
+  spawnMsg.set_initlon_(lon1.Degree());
+  spawnMsg.set_initz_(_depth);
+  spawnMsg.set_acommsaddress_(_acommsAddr);
+
+  _spawnPub.Publish(spawnMsg);
+}
+
+//////////////////////////////////////////////////
+TEST(SensorTest, TimeInterpolation)
+{
+  ignition::common::Console::SetVerbosity(4);
+
+  // Setup fixture
+  auto fixture = std::make_unique<ignition::gazebo::TestFixture>(
+      ignition::common::joinPaths(
+      std::string(PROJECT_SOURCE_PATH), "worlds", "empty_environment.sdf"));
+  unsigned int iterations{0u};
+
+  bool spawnedAllVehicles = {false};
+  static const std::string vehicles[] =
+    {"vehicle1"};
+
+  fixture->OnPostUpdate(
+    [&](const ignition::gazebo::UpdateInfo &_info,
+    const ignition::gazebo::EntityComponentManager &_ecm)
+    {
+      auto worldEntity = ignition::gazebo::worldEntity(_ecm);
+      ignition::gazebo::World world(worldEntity);
+      int numVehiclesSpawned{0};
+      for (auto v1 : vehicles)
+      {
+        if (world.ModelByName(_ecm, v1) != ignition::gazebo::kNullEntity)
+        {
+          numVehiclesSpawned++;
+        }
+      }
+
+      if (numVehiclesSpawned == 1)
+      {
+        spawnedAllVehicles = true;
+      }
+      iterations++;
+      duration = _info.simTime;
+    });
+  fixture->Finalize();
+
+  // Check that vehicles don't exist
+  fixture->Server()->RunOnce();
+  EXPECT_EQ(1, iterations);
+
+  int sleep{0};
+  int maxSleep{30};
+
+  // Change the dataconfig
+  igndbg << "Switching file" <<std::endl;
+  ignition::transport::Node node;
+  auto configPub = node.Advertise<ignition::msgs::StringMsg>(
+    "/world/science_sensor/environment_data_path");
+  ignition::msgs::StringMsg configMsg;
+  configMsg.set_data(ignition::common::joinPaths(
+      std::string(PROJECT_SOURCE_PATH), "data", "minimal_time_varying.csv"));
+  for (; !configPub.HasConnections() && sleep < maxSleep; ++sleep)
+  {
+    std::this_thread::sleep_for(100ms);
+  }
+  configPub.Publish(configMsg);
+
+  // Spawn first vehicle
+  auto spawnPub = node.Advertise<lrauv_ignition_plugins::msgs::LRAUVInit>(
+    "/lrauv/init");
+  for (; !spawnPub.HasConnections() && sleep < maxSleep; ++sleep)
+  {
+    std::this_thread::sleep_for(100ms);
+  }
+  ASSERT_LE(sleep, maxSleep);
+
+
+  // On a depth coplanar to a piece of data
+  SpawnVehicle(
+    spawnPub, "vehicle1", 0.000005, 0.000005, 5, 0);
+  node.Subscribe("/model/vehicle1/temperature", &TemperatureVeh1Cb);
+
+  // Check that vehicle was spawned
+  int expectedIterations = iterations;
+  for (sleep = 0; !spawnedAllVehicles && sleep < maxSleep;
+      ++sleep)
+  {
+    std::this_thread::sleep_for(100ms);
+    // Run paused so we avoid the physics moving the vehicles
+    fixture->Server()->RunOnce(true);
+    expectedIterations++;
+    igndbg << "Waiting for vehicles to spawn" << std::endl;
+  }
+  EXPECT_TRUE(spawnedAllVehicles);
+
+  fixture->Server()->Run(true, 1000, false);
```

Comment:
@arjo129 meta: there are a number of tools in `lrauv_system_tests` to reduce duplication, prevent races, reduce total run time, and improve readability across tests. I'd be inclined to use them if time allows. [`test_spawn.cc`](https://github.com/osrf/lrauv/blob/main/lrauv_system_tests/test/simulation/test_spawn.cc) is one good example.